app/assets/javascripts/admin/views/tasktype/task_type_create_view.js

```diff
@@ -1,175 +1,199 @@
-      isChecked(bool) {
-        return bool ? "checked" : "";
-      },
-      isSelected(bool) {
-        return bool ? "selected" : "";
-      },
-    };
-  }
+import { getTeams, createTaskType, updateTaskType, getTaskType } from "admin/admin_rest_api";
+import type { APITeamType } from "admin/api_flow_types";
 
-  initialize() {
-    this.isEditingMode = _.isString(this.model.id);
+const FormItem = Form.Item;
+const Option = Select.Option;
 
-    if (this.isEditingMode) {
-      this.listenTo(this.model, "sync", this.render);
-      this.model.fetch();
-    }
-  }
-
-  submitForm(event) {
-    event.preventDefault();
-
-    if (!this.ui.form[0].checkValidity()) {
-      Toast.error("Please supply all needed values.");
-      return;
-    }
+type Props = {
+  taskTypeId: ?string,
+  form: Object,
+};
 
-    const formValues = FormSyphon.serialize(this.ui.form);
-    if (formValues.settings.preferredMode === "Any") {
-      formValues.settings.preferredMode = null;
-    }
+type State = {
+  teams: Array<APITeamType>,
+};
 
-    // Add 'required' attribute to select once it's supported
-    // https://github.com/davidstutz/bootstrap-multiselect/issues/620
-    if (_.isEmpty(formValues.settings.allowedModes)) {
-      Toast.error("Please provide at least one allowed mode.");
-      return;
-    }
+class TaskTypeCreateView extends React.PureComponent<Props, State> {
+  state = {
+    teams: [],
+  };
 
-    this.model.save(formValues).then(() => app.router.navigate("/taskTypes", { trigger: true }));
+  componentDidMount() {
+    this.fetchData();
+    this.applyDefaults();
   }
 
-  onRender() {
-    const teamSelectionView = new SelectionView({
-      collection: new TeamCollection(),
-      childViewOptions: {
-        modelValue() {
-          return `${this.model.get("name")}`;
-        },
-        defaultItem: { name: this.model.get("team") },
+  async applyDefaults() {
+    const defaultValues = {
+      settings: {
+        somaClickingAllowed: true,
+        branchPointsAllowed: true,
+        advancedOptionsAllowed: true,
       },
-      data: "amIAnAdmin=true",
-      name: "team",
-      required: true,
-    });
-    this.showChildView("team", teamSelectionView);
+    };
+    const taskType = this.props.taskTypeId ? await getTaskType(this.props.taskTypeId) : null;
 
-    this.ui.multiselect.multiselect();
+    const formValues = Object.assign({}, defaultValues, taskType);
+    this.props.form.setFieldsValue(formValues);
   }
 
-  onBeforeDestroy() {
-    this.ui.multiselect.multiselect("destroy");
+  async fetchData() {
+    this.setState({ teams: await getTeams() });
+  }
+
+  handleSubmit = e => {
+    e.preventDefault();
+    this.props.form.validateFields((err, formValues) => {
+      if (!err) {
+        if (this.props.taskTypeId) {
+          updateTaskType(this.props.taskTypeId, formValues);
+        } else {
+          createTaskType(formValues);
+        }
+
+        app.router.navigate("/taskTypes", { trigger: true });
```

Comment:
Can we `await` `updateTaskType` and `createTaskType` before navigating back to taskTypes? Otherwise, errors will be silently swallowed.

Reply:
Good catch :-)